tests/basic/storage_pool_test.go

```diff
@@ -1408,16 +1415,52 @@ func addCloudDrive(stNode node.Node, poolID int32) error {
 	deviceSpec := driveSpecs[0]
 	deviceSpecParams := strings.Split(deviceSpec, ",")
 	var specSize uint64
+	var driveSize string
+
+	if poolID != -1 {
+		systemOpts := node.SystemctlOpts{
+			ConnectionOpts: node.ConnectionOpts{
+				Timeout:         2 * time.Minute,
+				TimeBeforeRetry: defaultRetryInterval,
+			},
+			Action: "start",
+		}
+		drivesMap, err := Inst().N.GetBlockDrives(stNode, systemOpts)
+		log.FailOnError(err, "error getting block drives from node %s", stNode.Name)
+
+	outer:
+		for _, v := range drivesMap {
+			labels := v.Labels
+			for _, pID := range labels {
+				if pID == fmt.Sprintf("%d", poolID) {
+					driveSize = v.Size
+					i := strings.Index(driveSize, "G")
+					driveSize = driveSize[:i]
+					break outer
+				}
+			}
+		}
+	}
+
+	paramsArr := make([]string, 0)
 	for _, param := range deviceSpecParams {
 		if strings.Contains(param, "size") {
 			val := strings.Split(param, "=")[1]
 			specSize, err = strconv.ParseUint(val, 10, 64)
 			if err != nil {
 				return fmt.Errorf("error converting size to uint64, err: %v", err)
 			}
+			log.FailOnError(err, "Error converting size to uint64")
```

Comment:
What is the point to use `log.FailOnError()` if `err` is not nil in the above statement, code will never hit this spot.
NOTE: Can we not mix `log.FailOnError()` and `return` in same function that has a return `err`, just return and handle it where you call this with `log.FailOnError()`, if need to

Reply:
fixed